hummingbot/wallet/ethereum/web3_wallet_backend.py

```diff
@@ -599,19 +602,19 @@ def _eth_unwrapped_event_listener(self, unwrapped_eth_event: WalletUnwrappedEthE
             tx_hashes.append(tx_hash)
         return tx_hashes
 
-    def wrap_eth(self, amount: float) -> str:
+    def wrap_eth(self, amount: Decimal) -> str:
         if self._weth_token is None:
             raise EnvironmentError("No WETH token address was used to initialize this wallet.")
 
         contract_func = self._weth_token.contract.functions.deposit()
         self.logger().info(f"Wrapping {amount} ether from wallet address {self.address}.")
-        return self.execute_transaction(contract_func, value=int(amount * 1e18))
+        return self.execute_transaction(contract_func, value=int(amount * 10**18))
```

Comment:
Ditto.

btw... this raw <-> nominal conversion seems to happen commonly enough that there should be some shared `@staticmethod` to do it.